server/server.js

```diff
@@ -62,11 +63,16 @@ app.use((req, res, next) => {
   next();
 });
 
-app.get('/', async (req, res) => {
-  res.render('main', { countDown: 300 - await userController.cachedCountDown.getCount() });
-});
+// caching the main page's response
+app.get(
+  '/',
+  responseCache(20 * 1000), // cache body data for 20 seconds
```

Comment:
these caching times are arbitory for when i was testing it. please review these times @Irev-Dev